### ISSUE TITLE
[core] Add `AppBar` and `Sidebar`

### ISSUE THREAD

**Issue by bharatkashyap** [mui/toolpad#4071]
- Add `AppBar` and `Sidebar` components for people who may wish to use only one instead of the entire `DashboardLayout`
- Move to this version of the component:
```tsx
 <DashboardLayout>
  <AppBar slots={{ actions: <CustomActions /> }} />
  <Sidebar />
  <PageContainer>
    <PageHeader slots={{toolbar: <CustomToolbar />}} />
      {children}
  </PageContainer>
</DashboardLayout>
```

- Benchmark: https://mui.com/material-ui/react-card/

### AppBar
![Screenshot 2024-09-10 at 2 20 45 PM](https://github.com/user-attachments/assets/01f33269-52b0-46c2-879b-19c5bc7d5146)
- Would have the `actions` and `account` slots 

### Drawer
![Screenshot 2024-09-10 at 2 21 05 PM](https://github.com/user-attachments/assets/c9920a63-cd3f-4a3d-bd72-46e9fd531b84)

- Would have the `footer` and `header` slots

The default `AppBar` and `Drawer` will ship as the default slots in `DashboardLayout`

Figma: 
AppBar: https://www.figma.com/design/14OEUnwh4ahMb2empYAEc8/Toolpad-UX?node-id=1540-1797&node-type=frame&t=mx15fYGux742bCAt-0

Drawer: https://www.figma.com/design/14OEUnwh4ahMb2empYAEc8/Toolpad-UX?node-id=1540-1816&node-type=frame&t=mx15fYGux742bCAt-0

**Search keywords**:

**Comment by prakhargupta1** [comment]
Like we have [hideNavigation](https://mui.com/toolpad/core/react-dashboard-layout/#hide-navigation) prop, maybe creating a similar hideAppBar prop can address some use cases where a user may want to hide it?

**Comment by prakhargupta1** [comment]
> What do you think about that?

I agree, making the dashbaordLayout composable through `appbar` and `sidebar` slots would make it more versatile. Let's  revisit it when we see more users requesting it.

**Comment by apedroferreira** [comment]
I already had an issue for the slots at https://github.com/mui/toolpad/issues/4299, but I agree with your more specific details about how to provide them.
I'll close the other issue and we can use this one, probably.

**Comment by sean-ideabank** [comment]
Wondering if there are any updates on this? I'm keen to use this feature to claim the header space for my pages

**Comment by bharatkashyap** [comment]
> Wondering if there are any updates on this? I'm keen to use this feature to claim the header space for my pages

Hey! Do you have a specific requirement that can't be met with the existing `DashboardLayout` right now? Feel free to create an issue describing that 👍🏻
